checkstyle-tester/diff.groovy

```diff
@@ -342,11 +342,11 @@ def getCommitSha(commitId, repoType, srcDestinationDir) {
     return sha.replace('\n', '')
 }
 
-def getCloneCmd(repoType, repoUrl, srcDestinationDir) {
+def getCloneCmd(repoType, repoUrl, srcDestinationDir, commitId) {
     def cloneCmd = ''
     switch (repoType) {
         case 'git':
-            cloneCmd = "git clone $repoUrl $srcDestinationDir"
+            cloneCmd = "git clone --depth 1 --branch $commitId $repoUrl $srcDestinationDir"
```

Comment:
I am not familiar with the nuances of —branch argument. What if the PR I am testing has multiple commits?

Reply:
Branch is label over top commit.
Commit is snapshot of whole repo (all files)

Reply:
Branch is just user friendly label to avoid dealing with long sha.
Branch, tag are just extra labels over tree of commits. 
Commits can make branch from trunk without  branch name, it is named headless branch.

Reply:
> --branch $commitId
> File format: REPO_NAME|[local|git|hg]|URL|[COMMIT_ID]|[EXCLUDE FOLDERS]

This doesn't work with the format of the file, as it doesn't allow commits

````
> git clone --depth 1 --branch ac6d6f093bc906839c682885b02ca1fd33e2f148 https://github.com/checkstyle/checkstyle.git
Cloning into 'checkstyle'...
warning: Could not find remote branch ac6d6f093bc906839c682885b02ca1fd33e2f148 to clone.
fatal: Remote branch ac6d6f093bc906839c682885b02ca1fd33e2f148 not found in upstream origin
````

Reply:
This will also fail if commits/branches change.

````
> git clone --depth 1 https://github.com/checkstyle/checkstyle.git                                                Cloning into 'checkstyle'...
remote: Enumerating objects: 7611, done.
remote: Counting objects: 100% (7611/7611), done.
remote: Compressing objects: 100% (5111/5111), done.
remote: Total 7611 (delta 3069), reused 4779 (delta 2279), pack-reused 0Receiving objects: 100% (7611/7611), 7.14 MiB | 7.08 MiB/s
Receiving objects: 100% (7611/7611), 7.25 MiB | 6.99 MiB/s, done.
Resolving deltas: 100% (3069/3069), done.
Updating files: 100% (6503/6503), done.
> cd .\checkstyle\
\checkstyle> git reset --hard ee921b6353824502ca46ed442f914f41128cea9b
fatal: Could not parse object 'ee921b6353824502ca46ed442f914f41128cea9b'.
````

Shallow clones are only good for CI, not for users.

Reply:
Here are nuances:
We use tags in almost all cases, tags usually doesn't change.
CI is most cases of usage now .
We rarely change tags in test projects, so unlikely need newer code.
Even we cloned in usual way to get new tag , we have to do git fetch, so reusage is highly unlikely.
It sources already exists on local, no clone happens, and we really do not much care on what version of sources we are running. 
Shallow is so quick, that it is easier to remove all , than think if your local sources are in good versions.

Reply:
>I think there is more benefits for us if we starts to use shallow.

I agree, but I think we need more testing. https://github.com/checkstyle/contribution/pull/819#discussion_r1421675065

We should update github action in main repo to make this easier.

Reply:
I shared a link to stackoverflow where people proves that it works.
And I shared testing on my local.

Reply:
I see no evidence of testing that these changes work with a projects file that uses commit SHA, for example: https://github.com/checkstyle/contribution/blob/52b49424115bbdcfe148b348118cd4a8df05d272/checkstyle-tester/projects-to-test-on.properties#L30. If there is, please point to it.


Again, https://github.com/checkstyle/contribution/pull/819#discussion_r1421675065

Reply:
support for sha was addded